arm-resourcehealth/2017-07-01/swagger/resourcehealth.json

```diff
@@ -0,0 +1,591 @@
+          },
+          "description": "The list of availabilityStatuses."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of availabilityStatuses. Call ListNext() with this URI to fetch the next page of availabilityStatuses."
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List availabilityStatus operation response."
+    },
+    "availabilityStatus": {
+      "type": "object",
+      "description": "availabilityStatus of a resource.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Azure Resource Manager Identity for the availabilityStatuses resource."
+        },
+        "name": {
+          "type": "string",
+          "description": "current."
+        },
+        "type": {
+          "type": "string",
+          "description": "Microsoft.ResourceHealth/AvailabilityStatuses."
+        },
+        "location": {
+          "type": "string",
+          "description": "Azure Resource Manager geo location of the resource."
+        },
+        "properties": {
+          "type": "object",
+          "description": "Properties of availability state.",
+          "properties": {
+            "availabilityState": {
+              "type": "string",
+              "description": "Availability status of the resource. When its null this availabilityStatus object represents an availability impacting event",
+              "enum": [
+                "Available",
+                "Unavailable",
+                "Unknown"
+              ],
+              "x-ms-enum": {
+                "name": "AvailabilityStateValues",
+                "modelAsString": false
+              }
+            },
+            "summary": {
+              "type": "string",
+              "description": "Summary description of the availability status."
+            },
+            "detailedStatus": {
+              "type": "string",
+              "description": "Details of the availability status."
+            },
+            "reasonType": {
+              "type": "string",
+              "description": "When the resource's availabilityState is Unavailable, it describes where the health impacting event was originated. Examples are planned, unplanned, user initiated or an outage etc."
+            },
+            "rootCauseAttributionTime": {
+              "type": "string",
+              "description": "When the resource's availabilityState is Unavailable, it provides the Timestamp for when the health impacting event was received.",
+              "format": "date-time"
+            },
+            "healthEventType": {
+              "type": "string",
+              "description": "In case of an availability impacting event, it describes when the health impacting event was originated. Examples are Lifecycle, Downtime, Fault Analysis etc."
+            },
+            "healthEventCause": {
```

Comment:
Yes, these and other properties which are present since version 1 are read only. Is there any special guidance for read-only properties? If so, we would be glad to get to them in next api-version.
--